src/components/forms/SubmissionView.js

```diff
@@ -1298,9 +1297,15 @@ export default class SubmissionView extends React.PureComponent{
                                 this.setState(stateToSet);
                             }
                         } else {
+                            // Check if parent validation state will change based on current submission... update that alongside rest of state, if so
+                            const newParentValidState = SubmissionView.findValidationState(parentKey, stateToSet.keyHierarchy, stateToSet.keyContext, stateToSet.keyComplete);
+                            if (newParentValidState !== stateToSet.keyValid[parentKey]) {
```

Comment:
Theoretically the 'if' check is not needed since if remains same, then.. nothing really differs upstream. PureComponent/React.memoized components will see the same value and not react to it.